optimize/backend/src/test/java/io/camunda/optimize/service/util/BpmnModelUtilTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.service.util;
+
+import static io.camunda.optimize.service.util.BpmnModelUtil.parseBpmnModel;
+import static io.camunda.optimize.util.ZeebeBpmnModels.END_EVENT;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SERVICE_TASK;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_CATCH;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_GATEWAY_CATCH;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_PROCESS_END;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_PROCESS_FIRST_SIGNAL;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_PROCESS_SECOND_SIGNAL;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_PROCESS_THIRD_SIGNAL;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_START_EVENT;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_START_INT_SUB_PROCESS;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SIGNAL_START_NON_INT_SUB_PROCESS;
+import static io.camunda.optimize.util.ZeebeBpmnModels.START_EVENT;
+import static io.camunda.optimize.util.ZeebeBpmnModels.USER_TASK;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createProcessWith83SignalEvents;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createSimpleServiceTaskProcess;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createSimpleUserTaskProcess;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.optimize.dto.optimize.FlowNodeDataDto;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.assertj.core.groups.Tuple;
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.camunda.bpm.model.bpmn.instance.EndEvent;
+import org.camunda.bpm.model.bpmn.instance.IntermediateCatchEvent;
+import org.camunda.bpm.model.bpmn.instance.Signal;
+import org.camunda.bpm.model.bpmn.instance.StartEvent;
+import org.camunda.bpm.model.xml.ModelInstance;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.junit.jupiter.api.Test;
+
+public class BpmnModelUtilTest {
+
+  public static final String CUSTOMER_ONBOARDING = "CustomerOnboarding";
+  public static final String END_EVENT_TYPE = "endEvent";
+  public static final String SERVICE_TASK_TYPE = "serviceTask";
+
+  public static final String SIMPLE_SERVICE_TASK_PROCESS =
+      Bpmn.convertToString(createSimpleServiceTaskProcess(CUSTOMER_ONBOARDING));
+
+  public static final String PROCESS_WITH_83_SIGNAL_EVENTS =
```

Comment:
```suggestion
  public static final String PROCESS_WITH_EMBEDDED_SUBPROCESS =
```

For this test, I think the subprocess is the important detail

Reply:
fixed